src/parser/meta.rs

```diff
@@ -0,0 +1,813 @@
+use nom::{
+    branch::alt,
+    bytes::streaming::{tag, take, take_while1},
+    character::{
+        complete::digit1,
+        streaming::{crlf, space0, space1},
+    },
+    combinator::{map, map_res, value},
+    multi::many0,
+    sequence::{pair, preceded, terminated, tuple},
+    IResult,
+};
+
+use super::{parse_u32, ErrorKind, MetaValue, Response, Status, Value};
+
+// TODO: remove this later in favour of individual parse_meta_*_status methods
+#[allow(dead_code)]
+pub fn parse_meta_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    terminated(
+        alt((
+            value(Response::Status(Status::NotStored), tag(b"NS")),
+            value(Response::Status(Status::Deleted), tag(b"DE")),
+            value(Response::Status(Status::Touched), tag(b"TO")),
+            value(Response::Status(Status::Exists), tag(b"EX")),
+            value(Response::Status(Status::NotFound), tag(b"NF")),
+        )),
+        crlf,
+    )(buf)
+}
+
+pub fn parse_meta_set_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    alt((
+        value(Response::Status(Status::Stored), tag(b"HD")),
+        value(Response::Status(Status::NotStored), tag(b"NS")),
+        value(Response::Status(Status::Exists), tag(b"EX")),
+        value(Response::Status(Status::NotFound), tag(b"NF")),
+    ))(buf)
+}
+
+pub fn parse_meta_get_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    alt((
+        value(Response::Status(Status::Value), tag(b"VA ")),
+        value(Response::Status(Status::Exists), tag(b"HD")),
+        value(Response::Status(Status::NotFound), tag(b"EN")),
+    ))(buf)
+}
+
+// TODO: refactor this into individual parse_meta_*_response methods
+#[allow(dead_code)]
+pub fn parse_meta_response(buf: &[u8]) -> Result<Option<(usize, Response)>, ErrorKind> {
+    let bufn = buf.len();
+    // TODO: alt calls parsers sequentially until one succeeds, which is not optimal.  Want to only call the correct parser, no sequencing.
```

Comment:
Yeah, there will be a small amount of code duplication for a _minor_ performance optimization since we won't have to do the fallthrough processing in `nom`'s `alt()` method.